command/agent/cache/lease_cache_test.go

```diff
@@ -726,59 +732,81 @@ func TestLeaseCache_PersistAndRestore(t *testing.T) {
+		expectMissingAfterRestore bool // If true, the response is not expected to be present in the restored cache
 	}{
 		{
 			// Make a request. A response with a new token is returned to the
 			// lease cache and that will be cached.
-			token:          "autoauthtoken",
-			method:         "GET",
-			urlPath:        "http://example.com/v1/sample/api",
-			body:           `{"value": "input"}`,
-			wantStatusCode: responses[0].Response.StatusCode,
+			token:   "autoauthtoken",
+			method:  "GET",
+			urlPath: "http://example.com/v1/sample/api",
+			body:    `{"value": "input"}`,
 		},
 		{
 			// Modify the request a little bit to ensure the second response is
 			// returned to the lease cache.
-			token:          "autoauthtoken",
-			method:         "GET",
-			urlPath:        "http://example.com/v1/sample/api",
-			body:           `{"value": "input_changed"}`,
-			wantStatusCode: responses[1].Response.StatusCode,
+			token:   "autoauthtoken",
+			method:  "GET",
+			urlPath: "http://example.com/v1/sample/api",
+			body:    `{"value": "input_changed"}`,
 		},
 		{
 			// Simulate an approle login to get another token
-			method:         "PUT",
-			urlPath:        "http://example.com/v1/auth/approle/login",
-			body:           `{"role_id": "my role", "secret_id": "my secret"}`,
-			wantStatusCode: responses[2].Response.StatusCode,
+			method:                    "PUT",
+			urlPath:                   "http://example.com/v1/auth/approle-expect-missing/login",
+			body:                      `{"role_id": "my role", "secret_id": "my secret"}`,
+			deleteFromPersistentStore: true,
+			expectMissingAfterRestore: true,
 		},
 		{
 			// Test caching with the token acquired from the approle login
-			token:          "testtoken2",
-			method:         "GET",
-			urlPath:        "http://example.com/v1/sample2/api",
-			body:           `{"second": "input"}`,
-			wantStatusCode: responses[3].Response.StatusCode,
+			token:   "testtoken2",
+			method:  "GET",
+			urlPath: "http://example.com/v1/sample-expect-missing/api",
+			body:    `{"second": "input"}`,
+			// This will be missing from the restored cache because its parent token was deleted
+			expectMissingAfterRestore: true,
+		},
+		{
+			// Simulate another approle login to get another token
+			method:  "PUT",
+			urlPath: "http://example.com/v1/auth/approle/login",
+			body:    `{"role_id": "my role", "secret_id": "my secret"}`,
+		},
+		{
+			// Test caching with the token acquired from the latest approle login
+			token:   "testtoken3",
+			method:  "GET",
+			urlPath: "http://example.com/v1/sample3/api",
+			body:    `{"third": "input"}`,
 		},
 	}
 
-	for _, ct := range cacheTests {
+	var deleteID string
+	for i, ct := range cacheTests {
 		// Send once to cache
 		sendReq := &SendRequest{
 			Token:   ct.token,
 			Request: httptest.NewRequest(ct.method, ct.urlPath, strings.NewReader(ct.body)),
 		}
+		if ct.deleteFromPersistentStore {
+			deleteID, err = computeIndexID(sendReq)
```

Comment:
I wonder if we check to see if we hare already set `deleteID` since it is possible to have more than one cache test with `deleteFromPersistentStore` set to true.

We may also  want to ensure that `deleteID` is not an empty string.